### PR TITLE
EWL-7199: SG2 | Create hero organism for Evergreen Page

### PR DESCRIPTION
**Pull request by caharder**
## Ticket(s)

**Jira Ticket**
- [EWL-7199: SG2 | Create hero organism for Evergreen Page](https://issues.ama-assn.org/browse/EWL-7199)

## Description
Adds hero evergreen molecule to SG for evergreen pages

## To Test
- [ ] Pull branch /feature/EWL-7199-hero-organism-Evergreen-Page
- [ ] run gulp serve
- [ ] visit [http://localhost:3000/?p=molecules-hero-evergreen](http://localhost:3000/?p=molecules-hero-evergreen)
- [ ] Verify that new hero is formatted as seen in screenshot below

## Visual Regressions
Checkout backstop VRT here
http://htmlpreview.github.io/?https://github.com/AmericanMedicalAssociation/ama-style-guide-2/blob/visual-regression-testing-artifact/feature/EWL-7199-hero-organism-Evergreen-Page/html_report/index.html

## Relevant Screenshots/GIFs
Screen Shot 2019-05-30 at 11 38 19 AM

## Remaining Tasks
- Add paragraph bundle for hero evergreen
- Add twig template for hero evergreen

## Additional Notes
N/A

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

**Comment by designerneil** [comment]
@caharder where are the variants for the hero?

Main Hero

Hero has text area on left with background color
Hero has 2 horizontal CTAs on the right with image behind
Hero stacks on mobile
Variant 1

Hero has text area on left with background color
Hero has 2 vertical CTAs on the right with image behind
Hero stacks on mobile
Variant 2

Hero has full width text area with background color

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: should be 
&__mobile-image {

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: &__aside {

File: styleguide/source/_patterns/02-molecules/hero-evergreen.twig
Comment (designerneil) [inline styleguide/source/_patterns/02-molecules/hero-evergreen.twig]: dont need `.exists`. If it's there it exists

**Review by designerneil — CHANGES_REQUESTED** [review]

**Comment by designerneil** [comment]
differences

File: styleguide/source/_patterns/02-molecules/hero-evergreen.twig
Comment (caharder) [reply styleguide/source/_patterns/02-molecules/hero-evergreen.twig]: Removed

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: Changed

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: Changed

**Comment by caharder** [comment]
Added variations:
- [No buttons](http://localhost:3000/?p=molecules-hero-evergreen)
- [2 buttons - horizontal](http://localhost:3000/?p=molecules-hero-evergreen-as-double-button)
- [2 buttons - stacked](http://localhost:3000/?p=molecules-hero-evergreen-as-stacked-button)
- [Single button](http://localhost:3000/?p=molecules-hero-evergreen-as-single-button)
- [Full width](http://localhost:3000/?p=molecules-hero-evergreen-as-full)
- [All Variations](http://localhost:3000/?p=viewall-molecules-hero-evergreen)

Also reduced padding on left container.

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: @include breakpoint($bp-large min-width) {
  background: none;
}

let's not use importants

If you use the mobile first approach then the last breakpoint would be large

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss:47]: If your already adding styling for buttons can you match the padding from the zeplins?

File: styleguide/source/_patterns/02-molecules/hero-evergreen/hero-evergreen.twig
Comment (designerneil) [inline styleguide/source/_patterns/02-molecules/hero-evergreen/hero-evergreen.twig:18]: needs spacec before endif

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss:5]: $bp-med min-width

**Review by designerneil — CHANGES_REQUESTED** [review]
Overall good job on covering all my change requests. I just have some minor updates needed before this gets merged.

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: Unfortunately, this doesn't do what I need it to do. The background image is set inline in the twig template, but I need `background: none` only on small size. The only way that I know of to override the inline CSS is to add `!important`. I used the `breakpoint($bp-small max-width)` because going with a mobile first approach, I would not be able to then override the `!important` needed for the small size (the background image is set in the template and I cannot pull that into the CSS). Do you know of a better way to accomplish this?

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss:47]: Done. I also reduced the font weight of the buttons per the zeplins.

File: styleguide/source/_patterns/02-molecules/hero-evergreen/hero-evergreen.twig
Comment (caharder) [reply styleguide/source/_patterns/02-molecules/hero-evergreen/hero-evergreen.twig:18]: Done.

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss:5]: Added.

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: There are 2 padding-tops here

padding-top: $gutter;
padding-top: 15px;

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [inline styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: This should be 0

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (designerneil) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: Instead of passing the background directly into the element add a class to it. Then in the twig template set it like so 
<style>
.backgroundClass {
background: transparent url("background-image: url( http://via.placeholder.com/800x600x186?l=3:2|800x600&s=36 );") 0 0  no-repeat;
}
</style>

**Review by designerneil — CHANGES_REQUESTED** [review]

**Comment by designerneil** [comment]
@caharder just a couple of changes needed before I push the merge button.

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: Great solution! It is now implemented.

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: This is corrected.

File: styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss
Comment (caharder) [reply styleguide/source/assets/scss/02-molecules/_hero-evergreen.scss]: Done.

**Review by designerneil — APPROVED** [review]

**Pull request merged**
